### PR TITLE
test coverage for ledger/eval.go funcs

### PR DESCRIPTION
**Pull request by brianolson**
## Summary

Test coverage for some functions in ledger/eval.go

## Test Plan

This is tests!

### CODE REVIEW

**Comment by brianolson** [comment]
roundCowState.compactCert() // 0% -> 100%
BlockEvaluator.TestTransactionGroup() // 0% -> 100%
BlockEvaluator.testTransaction() // 0% -> 100%
BlockEvaluator.transactionGroup() // 4 error cases not covered -> 1 (ErrNoSpace block full, tested in full system deploy tests)

BlockEvaluator.workaroundOverspentRewards() // testnet hotfix for rounds long gone, delete code?
BlockEvaluator.compactCertVotersAndTotal() // first two return clauses covered, nothing later, would need lots of setup to test.

**Comment by tsachiherman** [comment]
regarding `BlockEvaluator.workaroundOverspentRewards() // testnet hotfix for rounds long gone, delete code?` - 
unless we reset testnet, we won't be able to delete this. it's being used during catchup.

File: ledger/eval_test.go [hunk 1]
```diff
@@ -49,6 +51,14 @@ func init() {
...
 }
 
+func gadify(txgroup []transactions.SignedTxn) []transactions.SignedTxnWithAD {
+	txgroupad := make([]transactions.SignedTxnWithAD, len(txgroup))
+	for i, tx := range txgroup {
+		txgroupad[i].SignedTxn = tx
+	}
+	return txgroupad
+}
+
```
Comment (tsachiherman) [inline ledger/eval_test.go:61]: This method does not belong here. In fact, it should be placed (maybe) in bookkeeping, as the same logic is also used in `addToPendingBlockEvaluatorOnce` : 
```golang
	txgroupad := make([]transactions.SignedTxnWithAD, len(txgroup))
	for i, tx := range txgroup {
		txgroupad[i].SignedTxn = tx
	}
```

**Comment by tsachiherman** [comment]
Could you please open individual issues for the remaining cases you've not covered here ? ( i.e. with the same amount of details you've listed above )

File: ledger/eval_test.go [hunk 2]
```diff
@@ -572,3 +682,87 @@ func benchmarkBlockEvaluator(b *testing.B, inMem bool, withCrypto bool) {
...
+// a couple trivial tests that don't need setup
+// see TestBlockEvaluator for more
+func TestTestTransactionGroup(t *testing.T) {
+	var txgroup []transactions.SignedTxn
+	eval := BlockEvaluator{}
+	err := eval.TestTransactionGroup(txgroup)
+	require.NoError(t, err) // nothing to do, no problem
+
+	eval.proto.MaxTxGroupSize = 16
```
Comment (tsachiherman) [inline ledger/eval_test.go:750]: nit: use the current version to initialize the `proto` and set the txgroup size to `eval.proto.MaxTxGroupSize+1`

File: ledger/eval_test.go [hunk 3]
```diff
@@ -85,6 +95,52 @@ func TestBlockEvaluator(t *testing.T) {
...
+	// mixed fields should fail
+	btxn = txn
+	btxn.XferAsset = 3
+	st = btxn.Sign(keys[0])
+	txgroup = []transactions.SignedTxn{st}
+	err = eval.TestTransactionGroup(txgroup)
+	require.Error(t, err)
+	// Because they fail TestTransactionGroup they won't get added to the txn pool and so they won't actually be run; actually running the txn doesn't check txn.WellFormed()
+	// err = eval.Transaction(st, transactions.ApplyData{})
+	// require.Error(t, err)
```
Comment (algonautshant) [inline ledger/eval_test.go:142]: Can you please explain this? I am not able to understand it.
Also, can you please make the line shorter?

**Review by algonautshant — DISMISSED** [review]
Looks great.

**Review by tsachiherman — APPROVED** [review]
looks good to me.

File: ledger/eval_test.go (hunk 3 above)
Comment (brianolson) [reply ledger/eval_test.go]: reworded comment. hopefully clearer.

**Pull request merged**